rest/route/user_routes.go

```diff
@@ -91,3 +104,46 @@ func (h *userSettingsHandler) Execute(ctx context.Context, sc data.Connector) (R
 		Result: []model.Model{},
 	}, nil
 }
+
+func applyChanges(current user.UserSettings, changes model.APIUserSettings) (model.APIUserSettings, error) {
+	oldSettings := model.APIUserSettings{}
+	err := oldSettings.BuildFromService(current)
+	if err != nil {
+		return oldSettings, err
+	}
+
+	reflectOldSettings := reflect.ValueOf(&oldSettings)
+	reflectNewSettings := reflect.ValueOf(changes)
+	for i := 0; i < reflectNewSettings.NumField(); i++ {
+		propName := reflectNewSettings.Type().Field(i).Name
+		changedVal := reflectNewSettings.FieldByName(propName)
+		if changedVal.IsNil() {
+			continue
+		}
+		reflectOldSettings.Elem().FieldByName(propName).Set(changedVal)
```

Comment:
If types change this would need to be changed as well otherwise it won't compile (and the reflection should still work). I moved it to the rest model though